internal/security/svid/svid.go

```diff
@@ -0,0 +1,119 @@
+// Package svid provides functions to generate and sign svid.
+package svid
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"math/big"
+	"net/url"
+
+	"time"
+)
+
+// GenSVID generates SVID certificate for user and signs it based on given signing cert/key and public key algorithm
+func GenSVID(id string, expireInDays int, signingCert *x509.Certificate, signingKey any, keyAlgo x509.PublicKeyAlgorithm) (*tls.Certificate, error) {
```

Comment:
Since we are generating SPIFFE-ID SVID. I recommend we clearly separate the value of `CommanName` and `URIs`. 

e.g. For CommonName, we can use `usernamefoo`, for the `URI[0]` we can use `spiffe://test-abc.foo.bar/xyz/admin`

cc: @morrowc